### PR TITLE
refactor(creatable-select-field): Migrate to typescript

### PR DESCRIPTION
**Pull request by Rhotimee**
#### Summary
Migrate creatable-select-field to typescript
Update types for creatable-select-input

### CODE REVIEW

File: packages/components/fields/creatable-select-field/src/creatable-select-field.tsx [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+   */
+  hasWarning?: boolean;
+  /**
+   * Maximum height of the menu before scrolling
+   * <br>
+   * [Props from React select was used](https://react-select.com/props#creatable-props)
+   */
+  maxMenuHeight?: ReactSelectCreatableProps['maxMenuHeight'];
+  /**
+   * 	Dom element to portal the select menu to
```
Comment (CarlosCortizasCT) [inline packages/components/fields/creatable-select-field/src/creatable-select-field.tsx:179]:
```suggestion
   * Dom element to portal the select menu to
```

Remove extra spaces.

File: packages/components/fields/creatable-select-field/src/creatable-select-field.tsx [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+   */
+  menuPortalZIndex?: number;
+  /**
+   * whether the menu should block scroll while open
+   * <br>
+   * [Props from React select was used](https://react-select.com/props#creatable-props)
+   */
+  menuShouldBlockScroll?: ReactSelectCreatableProps['menuShouldBlockScroll'];
+  /**
+   * 	Name of the HTML Input (optional - without this, no input will be rendered)
```
Comment (CarlosCortizasCT) [inline packages/components/fields/creatable-select-field/src/creatable-select-field.tsx:195]:
```suggestion
   * Name of the HTML Input (optional - without this, no input will be rendered)
```
Remove extra spaces.

File: packages/components/fields/creatable-select-field/src/creatable-select-field.tsx [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+   * <br>
+   * [Props from React select was used](https://react-select.com/props#creatable-props)
+   */
+  name?: ReactSelectCreatableProps['name'];
+  /**
+   * Can be used to render a custom value when there are no options (either because of no search results, or all options have been used, or there were none in the first place).
+   * <br/>
+   * Gets called with `{ inputValue: String }`. `inputValue` will be an empty string when no search text is present.
+   * <br>
```
Comment (CarlosCortizasCT) [inline packages/components/fields/creatable-select-field/src/creatable-select-field.tsx:204]:
```suggestion
   * <br/>
```

File: packages/components/fields/creatable-select-field/src/creatable-select-field.tsx [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+   * [Props from React select was used](https://react-select.com/props#creatable-props)
+   */
+  noOptionsMessage?: ReactSelectCreatableProps['noOptionsMessage'];
+  /**
+   * Handle blur events on the control
+   */
+  onBlur?: (event: TEvent) => void;
+  /**
+   * Called with a fake event when value changes.
+   * <br />
```
Comment (CarlosCortizasCT) [inline packages/components/fields/creatable-select-field/src/creatable-select-field.tsx:214]: nit: We have `<br>`, `<br/>` and `<br />` across the props.
Maybe we could standarize it.

**Review by CarlosCortizasCT — APPROVED** [review]
LGTM

**Review by ddouglasz — APPROVED** [review]
LGTM

**Pull request merged**